lib/assessment.py

```diff
@@ -1,80 +1,89 @@
+#!/usr/bin/env python3
+
 import re
 import sys
 import fileinput
 from lxml import etree
-import ConfigParser
 import os
 import importlib
 import time
 import logging
+import yaml
+import subprocess
 
-external_tests_path=""
-internal_tests_path=""
-assessment_settings = ConfigParser.ConfigParser()
-assessment_settings.read(os.environ.get('SECANT_CONFIG_DIR', '/etc/secant') + '/' + 'assessment.conf')
-secant_settings = ConfigParser.ConfigParser()
-if os.path.split(os.getcwd())[-1] == 'lib':
-    sys.path.append('../include')
-    external_tests_path="../external_tests"
-    internal_tests_path="../internal_tests"
-else:
-    sys.path.append('include')
-    external_tests_path="external_tests"
-    internal_tests_path="internal_tests"
-
-import py_functions
+dirname = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(0, dirname+'/../include/')
+probes_path='/../probes/'
+conf_path = dirname + '/../conf/secant.conf'
+from py_functions import getSettingsFromBashConfFile
 
-py_functions.setLogging()
-
-def assessment(template_id, report_file, tests_version, base_mpuri):
+def assessment(template_id, report_file, tests_version, base_mpuri, message_id):
     total_outcome = False
     secant =  etree.Element('SECANT')
     version = etree.SubElement(secant, "VERSION")
     imageID = etree.SubElement(secant, "IMAGEID")
     date = etree.SubElement(secant, "DATE")
     outcome = etree.SubElement(secant, "OUTCOME")
+    outcome_description = etree.SubElement(secant, "OUTCOME_DESCRIPTION")
+    messageID = etree.SubElement(secant, "MESSAGEID")
     log = etree.SubElement(secant, "LOG")
     version.text = tests_version
     imageID.text = base_mpuri
-
+    messageID.text = message_id
     report = etree.parse(report_file)
 
-    for tests_type in (external_tests_path, internal_tests_path):
-        for dir_name in os.listdir(tests_type):
-            test_name = dir_name.upper()
-            if report.find(test_name) is None:
-                continue
-
+    total_outcomeE = False
+    total_outcomeI = False
+    config = getSettingsFromBashConfFile(conf_path, 'SECANT_PROBES').split(',')
+    for probe_name in config:
+        if report.find(probe_name) is not None:
```

Comment:
If the report doesn't exist, it means that something very fishy is going on. It's better to immediately stop and issue an exception or error code.